dask/array/core.py

```diff
@@ -1500,12 +1500,21 @@ def __rxor__(self, other):
     def __matmul__(self, other):
         if not hasattr(other, 'ndim'):
             other = np.asarray(other)  # account for array-like RHS
+        if other.ndim > 2:
+            msg = ('Matrix multiplication for higher-dimensional arrays '
+                   'is not yet implemented in Dask. See the discussion at '
+                   'https://github.com/dask/dask/pull/2349 for details.')
+            raise NotImplementedError(msg)
```

Comment:
Yeah, this makes it sound like dask.arrays haven't implemented higher dimensional tensordot operations.  It would be good to specify that the limitation is just with the matmul operator syntax.  We should probably point the user towards the full `tensordot` function instead.